src/lib/mina_net2/tests/tests.ml

```diff
@@ -183,11 +178,18 @@ let%test_module "coda network tests" =
           open_stream c ~protocol:"echo" ~peer:b_peerid >>| Or_error.ok_exn
         in
         let r, w = Libp2p_stream.pipes stream in
+        let testmsg =
+          String.of_char_list
+          @@ Quickcheck.random_value
+               (Quickcheck.Generator.list_with_length 1_000_000
+                  Quickcheck.Generator.char_print )
+        in
+        let testmsg = String.concat (List.init 170 ~f:(const testmsg)) in
         Pipe.write_without_pushback w testmsg ;
         Pipe.close w ;
         (* HACK: let our messages send before we reset.
            It would be more principled to add synchronization. *)
-        let%bind () = after (Time.Span.of_sec 1.) in
+        let%bind () = after (Time.Span.of_sec 120.) in
```

Comment:
Is a 2 minute timeout really necessary here?

Reply:
When everything is ok, this timeout is not hit.

It should be more than `1s` for sure because now this test attempts to process many megabytes of data and `1s` is simply not enough. I've set `120s` just to be large-enough-not-to-care. I can try to set a smaller value, but once again, it will matter only at a point we break something.

Reply:
Nit: use an `Ivar.t` to signal synchronization instead of waiting.

Reply:
It's a bit less simple than that, but I fixed the test to not to rely on `time` (`after` is now used only for timing out)